### PR TITLE
Database connection config

### PR DESCRIPTION
**Pull request by caroo7**
- hibernate configuration (by adding spring beans)
- test connection to postgres database using simple 'book' entity
- configure hsql database in future test purposes

### CODE REVIEW

File: database/src/main/java/main/Main.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package main;
+
+import config.DatabaseConfiguration;
+import entities.Book;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import repositories.BookRepository;
+
+public class Main {
```
Comment (caroo7) [inline database/src/main/java/main/Main.java]: I removed this class in next commit - it's not necessary right now.

**Comment by LazyScoundrel** [comment]
Nothing to add from my side as this is just for configuration purposes. Eligible to merge :)

File: database/pom.xml [hunk 2]
```diff
@@ -14,4 +14,70 @@
...
+
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-context</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.springframework.data</groupId>
+            <artifactId>spring-data-jpa</artifactId>
+            <version>1.10.2.RELEASE</version>
```
Comment (LukaszCzubernat) [inline database/pom.xml]: It would be nice if You move those 'numbers' to properties

File: database/pom.xml [hunk 3]
```diff
@@ -14,4 +14,70 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>cglib</groupId>
+            <artifactId>cglib</artifactId>
+            <version>3.2.1</version>
```
Comment (LukaszCzubernat) [inline database/pom.xml]: Same here

File: database/pom.xml [hunk 4]
```diff
@@ -14,4 +14,70 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+            <version>9.1-901-1.jdbc4</version>
```
Comment (LukaszCzubernat) [inline database/pom.xml]: Again

File: database/pom.xml [hunk 5]
```diff
@@ -14,4 +14,70 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.hibernate.javax.persistence</groupId>
+            <artifactId>hibernate-jpa-2.1-api</artifactId>
+            <version>1.0.0.Final</version>
```
Comment (LukaszCzubernat) [inline database/pom.xml]: And again :)

File: database/pom.xml [hunk 6]
```diff
@@ -14,4 +14,70 @@
...
+
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-test</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.hsqldb</groupId>
+            <artifactId>hsqldb</artifactId>
+            <version>2.3.4</version>
```
Comment (LukaszCzubernat) [inline database/pom.xml]: You what to do :smile:

**Comment by LukaszCzubernat** [comment]
Nice and clean.
Nothing more to complain about

File: database/pom.xml [hunk 7]
```diff
@@ -14,4 +14,80 @@
     <name>database</name>
     <description>Database access layer. Contains all entities and DAOs.</description>
 
+    <properties>
+        <spring.data.jpa.version>1.10.2.RELEASE</spring.data.jpa.version>
+        <cglib.version>3.2.1</cglib.version>
+        <postgresql.version>9.1-901-1.jdbc4</postgresql.version>
+        <hibernate.jpa.version>1.0.0.Final</hibernate.jpa.version>
+        <hsqldb.version>2.3.4</hsqldb.version>
+    </properties>
```
Comment (LIttleAncientForestKami) [inline database/pom.xml:23]: :+1:

File: database/pom.xml [hunk 8]
```diff
@@ -14,4 +14,80 @@
...
+
+    <dependencies>
+
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-context</artifactId>
```
Comment (LIttleAncientForestKami) [inline database/pom.xml:29]: Is version defined someplace else? :eyes:

File: database/pom.xml [hunk 9]
```diff
@@ -14,4 +14,80 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-test</artifactId>
```
Comment (LIttleAncientForestKami) [inline database/pom.xml:81]: Scope test? Or defined elsewhere? :eyes:

File: database/pom.xml [hunk 10]
```diff
@@ -14,4 +14,80 @@
...
+
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
```
Comment (LIttleAncientForestKami) [inline database/pom.xml:76]: Scope test? Or defined elsewhere? :eyes:

File: database/src/main/java/entities/Book.java [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+
+    private String title;
+
+    private String authors;
+
+    private String description;
+
+    // add library list (entity), tag list (entity), category id (entity), discount type (enum)
+
+    private Book() {
```
Comment (LIttleAncientForestKami) [inline database/src/main/java/entities/Book.java:24]: Why not one line if empty private constructor?

File: database/pom.xml (hunk 8 above)
Comment (caroo7) [reply database/pom.xml:29]: Yep, in parent pom ;)

File: database/pom.xml (hunk 9 above)
Comment (caroo7) [reply database/pom.xml:81]: In parent pom we specified that testng is used only for tests.

File: database/src/main/java/entities/Book.java (hunk 11 above)
Comment (caroo7) [reply database/src/main/java/entities/Book.java:24]: In next pull request I will remove this constructor, cause there will be lombok annotation for this purpose.

**Pull request merged**